src/components/DevelopmentGuide/ConditionalViewport.mdx

```diff
@@ -0,0 +1,114 @@
+applied on Android, it prevents users from zooming in correctly at all (which is
+its own accessibility violation).
+
+So for everything else, the viewport tag should be:
+
+<Source
+  code={`
+<meta name="viewport" content="width=device-width">
+`}
+  language="html"
+/>
+
+Setting the viewport conditionally requires capturing whether the device is iOS or not.
+How you do this will depend on your app– the below approaches are for Next.js apps (App or Pages Router).
+
+### Approach for Next.js App Router apps
+
+Using the App router, you can customize the initial viewport with the dynamic [`generateViewport()`](https://nextjs.org/docs/app/api-reference/functions/generate-viewport) function. This only works in a server component, so most likely will be defined near the root of your project in
+the `page.tsx` or `layout.tsx` file.
+
+Then, you can access the [`userAgent`](https://nextjs.org/docs/app/api-reference/functions/userAgent) object from Next's [`headers()`](https://nextjs.org/docs/app/api-reference/functions/headers) function. If the current device is an iPhone, return the necessary viewport object.
+
+<Source
+  code={`
+
+// layout.tsx or page.tsx
+
+export async function generateViewport() {
+  const userAgent = headers().get("user-agent");
+  const isiPhone = /iphone/i.test(userAgent ?? "");
+  return isiPhone ? 
+  { width: "device-width", initialScale: 1, maximumScale: 1, userScalable: false }
+    : {};
+}
+    
+`}
+language="js"
+/>
+
+This approach is currently implemented on [Digital Collections](https://github.com/NYPL/digital-collections/blob/8758f1106af59c4a76015d978700ec8baba6deb4/app/layout.tsx#L45).
+
+### Approach for Next.js Pages Router apps
+
+In a Pages router app, you need to find the `userAgent` object from a client component (again, probably near the root of your project).
+There are more ways to do this (and many packages), but here's an option:
+
+<Source
+  code={`
+import { Helmet } from "react-helmet";
+
+const [isIOS, setIsIOS] = useState(false);
+
+useEffect(() => {
+  const userAgent = typeof window !== "undefined" && navigator.userAgent;
+  if (userAgent && /iPad|iPhone|iPod/.test(userAgent)) {
+    setIsIOS(true);
+  }
+}, []);
+
+const viewport = isIOS ?
+(<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>)
+: (<meta name="viewport" content="width=device-width" />);
+
+return(
+<>
+  <Helmet>{viewport}</Helmet>
+  ...
+</>
+) 
+`} language="js" />
+
+[`Helmet`](https://www.npmjs.com/package/react-helmet) is a React package that injects html into the `<header>` from the client component, necessary here because the `useEffect()` is reading
+`window.navigator` object to get `userAgent` above. Once the device is known, you set the state to `isIOS` and inject the viewport tag accordingly.
+
+This approach is currently implemented on [Turbine](https://github.com/NYPL/nypl-ds-test-app/blob/09258593a19e0650b1a3ddfddd4181a8096b796f/pages/_app.tsx#L60).
```

Comment:
Is this code not available yet on Github?  The link is throwing a 404 error.

Reply:
It's a private repo so maybe that's why?

Reply:
Link works for me now, lmk if it's still a 404– even if it's private, any of us should be able to see files in that repo

Reply:
Ah, my error.  It works fine when I open it in a browser where I am actually logged into github. :)